From: srinivas pulukuru <srinivas.pulukuru@ti.com>
Date: Fri, 6 May 2011 18:02:33 +0000 (-0500)
Subject: OMAP4 DSS HDMI: fix video_power mode for hdmi in suspend-resume
X-Git-Url: http://omapzoom.org/?p=kernel%2Fomap.git;a=commitdiff_plain;h=352c5e9cc35cfa5fe600d6cc8bb1226762340e1f

OMAP4 DSS HDMI: fix video_power mode for hdmi in suspend-resume

1. When going in suspend the video_power should be set HDMI_POWER_MIN
2. When resuming hdmi driver uses the video_power to start up in
   min power mode. this will setup the driver to handle the incoming
   phy connect / disconnect / first_hpd interrupts.
3. Currently the system is going to suspend with video_power unchanged
   causing it to resume in full or min  power mode based on the last
   state before suspend.
4. Hot plug status notification should only be sent after the phy is
   fully configured in full power mode and edid read successfully.

Change-Id: I58d24dbef3f8691525c133a45f278edc931fd35a
Signed-off-by: srinivas pulukuru <srinivas.pulukuru@ti.com>
---

diff --git a/drivers/video/omap2/dss/hdmi.c b/drivers/video/omap2/dss/hdmi.c
index c920da7..52a745b 100644
--- a/drivers/video/omap2/dss/hdmi.c
+++ b/drivers/video/omap2/dss/hdmi.c
@@ -1721,7 +1721,7 @@ err:
 static int hdmi_set_power(struct omap_dss_device *dssdev,
 	enum hdmi_power_state v_power, bool suspend, bool _audio_on)
 {
-	int r = 0, old_power;
+	int r = 0;
 	enum hdmi_power_state power_need;
 	enum hdmi_dev_state state_need;
 
@@ -1739,8 +1739,6 @@ static int hdmi_set_power(struct omap_dss_device *dssdev,
 		power_need, (state_need == HDMI_ACTIVE ? 'A' :
 			     state_need == HDMI_SUSPENDED ? 'S' : 'D'));
 
-	old_power = hdmi_power;
-
 	/* turn on HDMI */
 	if (dssdev->state != OMAP_DSS_DISPLAY_ACTIVE && power_need)
 		r = hdmi_start_display(dssdev);
@@ -1915,9 +1913,11 @@ static int hdmi_suspend_video(struct omap_dss_device *dssdev)
 	if (dssdev->state != OMAP_DSS_DISPLAY_ACTIVE) {
 		ret = -EINVAL;
 	} else {
-		/* (attempt to=ignore result) turn audio power off */
+		/* (attempt to=ignore result) turn audio power off and
+		 * set hdmi video_power to min
+		 */
 		hdmi_audio_power_off();
-		ret = hdmi_set_power(dssdev, video_power, true, audio_on);
+		ret = hdmi_set_power(dssdev, HDMI_POWER_MIN, true, audio_on);
 	}
 
 	mutex_unlock(&hdmi.lock_aux);
@@ -1944,12 +1944,6 @@ static int hdmi_resume_video(struct omap_dss_device *dssdev)
 
 		/* signal power change */
 		mutex_unlock(&hdmi.lock_aux);
-		if (!ret)
-			hdmi_notify_pwrchange(HDMI_EVENT_POWERON);
-
-		/* if edid is already read, give hot plug notification */
-		if (edid_set)
-			set_hdmi_hot_plug_status(dssdev, true);
 	}
 	mutex_unlock(&hdmi.lock);
 